### ISSUE TITLE
Repositories not showing langauge statistics

### ISSUE THREAD

**Issue by SuNNjek** [go-gitea/gitea#12020]
- Gitea version (or commit ref): 1.12.1
- Git version: -
- Operating system: Raspbian 10
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

Shortly after upgrading from 1.11.7 to 1.12.0 the web interface hung (I assume while analysing the languages used in my repos) so I decided to restart Gitea, which made it respond again. However, I just realized my older repositories don't show any language statistics at all, even though they should.
I don't know if there currently is a way to manually trigger another analysis (if there is I can't find it), but if there isn't, it would be nice if you could do that from the admin page.

**Comment by lafriks** [comment]
If no language stats are calculated for repository it should calculate automatically on Gitea restart

**Comment by SuNNjek** [comment]
@lafriks Having restarted my Gitea, I unfortunately cannot confirm this, the statistics are still missing

**Comment by lafriks** [comment]
Can you check does table `repo_indexer_status` has record for specific repository

**Comment by SuNNjek** [comment]
@lafriks No, there are too few entries in the table. I also checked a specific repo and there is no entry with the repo id in that table.

**Comment by SuNNjek** [comment]
@lafriks I just noticed something else. I don't know if this has anything to do with it, but for some reason on my dashboard I see this:
![grafik](https://user-images.githubusercontent.com/20151081/85528867-c0e02800-b60c-11ea-8d0b-5539802633e6.png)

Notice it says I have only 13 repos (the `repo_indexer_status` table also only has 13 entries :thinking:) when my user does in fact have 39 repositories (as it correctly says at the top)

**Comment by SuNNjek** [comment]
@lafriks so it seems these ominous 13 repos all have `is_archived=false` while the others have `is_archived=NULL` and therefore get filtered out. I assume this goes for the statistics too?

A bit of background info on my installation:
I'm running Gitea on a Raspberry Pi 4B running Raspbian 10 on Docker (self built image on the latest 1.12.1 tag). The repos that are missing info are all older from before the Pi 4 was released (and I was on a Pi 3B). I tried the backup/restore function to migrate to the new Pi, and I had some problems with some indexes if I recall correctly, so maybe this could be a consequence of this?
